### PR TITLE
TIED-54, 55 & 49: Contanerizing the app

### PR DESCRIPTION
**Pull request by arpargo**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -26,11 +26,23 @@ $ cd helerm-ui
 
 #### 2. Install depencies:
 
+1. Running the app locally
```
Comment (mikkojamG) [inline README.md:29]: I think the indentations / the order of steps looks a bit funny 🤔 Would suggest something like this:

### Install from source

Make sure you have [Yarn](https://yarnpkg.com/en/docs/install) installed globally.

#### 1. Clone the project:

```bash
$ git clone https://github.com/City-of-Helsinki/helerm-ui.git
$ cd helerm-ui
```

#### 2. Install depencies:

```bash
$ yarn              # Install project dependencies
```

#### 3. Running the app locally
```bash
$ yarn start:dev    # Launch on local environment
```

### Running the app in docker

#### 1. Add needed environmental variables
Add these by copying the `.env.example` to `.env` and adding the values to the file, or directly through process.env.

Requires you to setup a port where the application will be running to your .env file, f.ex.
```bash
PORT=3000
```

```bash
$ docker-compose up --build # Launches the containerized version of the application
```

You will need to have the CLIENT_* and JWT_TOKEN parameters for the Helsinki Tunnistamo, or **run your own https://github.com/City-of-Helsinki/tunnistamo instance**.

#### 2. Navigate to [http://localhost:3000](http://localhost:3000)

File: README.md [hunk 2]
```diff
@@ -24,15 +24,25 @@ $ git clone https://github.com/City-of-Helsinki/helerm-ui.git
...
 
-#### 2. Install depencies:
+#### 2. Install project dependencies
 
-1. Running the app locally
 ```bash
 $ yarn              # Install project dependencies
+```
+
+#### 3. Add needed environmental variables
```
Comment (mikkojamG) [inline README.md:33]: Great, looks a lot better 👍

File: Dockerfile [hunk 3]
```diff
@@ -0,0 +1,33 @@
+# ==========================================
+FROM helsinkitest/node:14-slim as deployable
+# ==========================================
+# Hardcode $HOME to /app, as yarn will need to put some stray files inside $HOME
+# In Openshift $HOME would be / by default
+ENV HOME /app
+# Most files from source tree are needed at runtime
+COPY  . .
```
Comment (danipran) [inline Dockerfile:8]: Double space between `COPY` and `.`. Not the most serious of offenses, but happened to catch my eyes 🙂

**Review by danipran — CHANGES_REQUESTED** [review]
Should also check what Sonarcloud is warning about

File: compose.yml [hunk 4]
```diff
@@ -0,0 +1,10 @@
+version: '3.8'
```
[the remark this answers is not in the document]
Comment (arpargo) [reply compose.yml:1]: According to most recent Docker documentation compose.yml is the preferred naming convention for docker compose files these days.

File: Dockerfile (hunk 3 above)
Comment (arpargo) [reply Dockerfile:8]: One of the issues Sonarcloud is complaining about, fixed on next commit which is focused on fixing Sonarcloud issues.

File: Dockerfile [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+
+# Official image has npm log verbosity as info. More info - https://github.com/nodejs/docker-node#verbosity
+ENV NPM_CONFIG_LOGLEVEL warn
+
+EXPOSE ${PORT}
+# Yarn
+ENV YARN_VERSION 1.19.1
```
[the remark this answers is not in the document]
Comment (arpargo) [reply Dockerfile:15]: Iirc it had something to do with libraries that had not been updated when this dockerfile was created. Could check if you can now use the latest with the node updated.

[the remark this answers is not in the document]
Comment (danipran) [reply Dockerfile:15]: Thanks!

File: compose.yml (hunk 4 above)
[the remark this answers is not in the document]
Comment (danipran) [reply compose.yml:1]: Oh! Didn't know, interesting 🤔 That is indeed true: https://docs.docker.com/compose/compose-file/#compose-file
Consider this resolved 👍

File: Dockerfile (hunk 5 above)
[the remark this answers is not in the document]
Comment (danipran) [reply Dockerfile:20]: Will be solved in a separate task.

**Review by danipran — APPROVED** [review]
Well, this PR got a bit messy due to other PRs wanting to merge into this PR's branch 😅 discussed this with Arto and gave tips how to avoid this in the future. LGTM otherwise :+1:

**Pull request merged**